### PR TITLE
Support properties in where clause

### PR DESCRIPTION
**Pull request by nevermore3**
#### What type of PR is this?
- [ ] bug
- [x] feature
- [ ] enhancement

#### What does this PR do?
1、support properties function in GO sentence 's  where clause

2、use DeducePropVisitor to extract the attribbute  of  VertexExpression & EdgeExpression

#### Which issue(s)/PR(s) this PR relates to?
https://github.com/vesoft-inc/nebula/issues/3117, will reconsider later
  
#### Checklist：
- [x] Documentation affected （Please add the label if documentation needs to be modified.)
- [ ] Incompatible （If it is incompatible, please describe it and add corresponding label.）
- [ ] Need to cherry-pick （If need to cherry-pick to some branches, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

#### Release notes：
Please confirm whether to reflect in release notes and how to describe:

### CODE REVIEW

File: tests/tck/features/go/GO.feature [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Copyright (c) 2021 vesoft inc. All rights reserved.
 #
 # This source code is licensed under Apache 2.0 License.
+@jmq
```
[the remark this answers is not in the document]
Comment (nevermore3) [reply tests/tck/features/go/GO.feature]: done

File: src/graph/visitor/DeducePropsVisitor.cpp [hunk 2]
```diff
@@ -173,17 +177,63 @@ void DeducePropsVisitor::visit(VersionedVariableExpression *expr) { reportError(
...
+  if (tagIds_ == nullptr) {
+    auto tagStatus = qctx_->schemaMng()->getAllLatestVerTagSchema(space_);
+    if (!tagStatus.ok()) {
+      status_ = std::move(tagStatus).status();
+      return;
+    }
+    for (const auto &tag : tagStatus.value()) {
+      tagIds.emplace_back(tag.first);
+    }
+    tagIds_ = &tagIds;
```
[the remark this answers is not in the document]
Comment (nevermore3) [reply src/graph/visitor/DeducePropsVisitor.cpp]: ok

File: src/graph/visitor/DeducePropsVisitor.cpp [hunk 3]
```diff
@@ -173,17 +177,66 @@ void DeducePropsVisitor::visit(VersionedVariableExpression *expr) { reportError(
 
 void DeducePropsVisitor::visit(LabelExpression *expr) { reportError(expr); }
 
-void DeducePropsVisitor::visit(AttributeExpression *expr) { UNUSED(expr); }
```
Comment (Shylock-Hg) [inline src/graph/visitor/DeducePropsVisitor.cpp:176]: Why delete this?

File: src/graph/visitor/DeducePropsVisitor.cpp [hunk 4]
```diff
@@ -173,17 +177,66 @@ void DeducePropsVisitor::visit(VersionedVariableExpression *expr) { reportError(
...
 
 void DeducePropsVisitor::visit(ConstantExpression *expr) { UNUSED(expr); }
 
-void DeducePropsVisitor::visit(VertexExpression *expr) { UNUSED(expr); }
+void DeducePropsVisitor::visit(ColumnExpression *expr) { UNUSED(expr); }
 
-void DeducePropsVisitor::visit(EdgeExpression *expr) { UNUSED(expr); }
+void DeducePropsVisitor::visit(VertexExpression *expr) {
+  std::vector<TagID> tagIds;
+  if (tagIds_ == nullptr) {
```
Comment (Shylock-Hg) [inline src/graph/visitor/DeducePropsVisitor.cpp:188]: Initialize `tagIds_` in constructor. Here will call just when meet `VertexExpression`.

File: src/graph/visitor/DeducePropsVisitor.cpp [hunk 5]
```diff
@@ -173,17 +177,66 @@ void DeducePropsVisitor::visit(VersionedVariableExpression *expr) { reportError(
...
+      for (size_t i = 0; i < tagSchema->getNumFields(); ++i) {
+        exprProps_->insertTagProp(tagID, tagSchema->getFieldName(i));
+      }
+    }
+  }
+}
 
-void DeducePropsVisitor::visit(ColumnExpression *expr) { UNUSED(expr); }
+void DeducePropsVisitor::visit(EdgeExpression *expr) {
+  if (edgeTypes_ == nullptr) {
```
Comment (Shylock-Hg) [inline src/graph/visitor/DeducePropsVisitor.cpp:225]: This behavior different from tag schema

File: src/graph/visitor/DeducePropsVisitor.cpp (hunk 3 above)
Comment (nevermore3) [reply src/graph/visitor/DeducePropsVisitor.cpp:176]: properties($$).age is AttributeExpression, we should extract attribute of Vertex

File: src/graph/visitor/DeducePropsVisitor.cpp (hunk 5 above)
Comment (nevermore3) [reply src/graph/visitor/DeducePropsVisitor.cpp:214]: fix

File: src/graph/visitor/DeducePropsVisitor.cpp (hunk 4 above)
Comment (nevermore3) [reply src/graph/visitor/DeducePropsVisitor.cpp:187]: fix

**Review by yixinglu — APPROVED** [review]
This `properties(*).age` will be updated into `properties(*).'player.age'`. LGTM

**Comment by Shylock-Hg** [comment]
Add a test case like `fetch prop on * yield vertex as v` and `fetch prop on <edge_type> yield edge as e`

**Comment by nevermore3** [comment]
> Add a test case like `fetch prop on * yield vertex as v` and `fetch prop on <edge_type> yield edge as e`. And lookup , find path and match maybe should test too.

these testcase already exist,  this PR is  support the properties function in where clause

**Review by Shylock-Hg — APPROVED** [review]

**Review by jievince — APPROVED** [review]
